cmake/DetectOptions.cmake

```diff
@@ -141,6 +141,16 @@ endif()
 
 set(mpi_find_components C)
 
+# Cuda
+if(ADIOS2_USE_CUDA STREQUAL AUTO)
+  find_package(CUDA)
+elseif(ADIOS2_USE_CUDA)
+  find_package(CUDA REQUIRED)
+endif()
+if(CUDA_FOUND)
+  set(ADIOS2_HAVE_CUDA TRUE)
+endif()
```

Comment:
`find_package(CUDA)` has been deprecated, and it's goal was to enable CUDA as a language, but what you need here is really just the toolkit. There is `find_package(CUDAToolkit)` now, but it's a relatively recent addition to cmake, so you might have to bump up the `cmake_minimum_required` to make sure it's available. In any case, you have cmake experts on your team, I'm sure they know how to handle this best ;)

Reply:
CUDA aware MPI relies on the Unified Virtual Addressing (UVA) feature introduced in CUDA 4.0 to detect if a buffer resides on the device or host. I chose to use a memory space because we might want to extend this to AMD for Frontier (with either HIP, Kokkos, etc) and I am not aware of any solution that can detect where a buffer has been allocated across all architectures.

Thanks for pointing me to `CUDAToolkit`. Will bring it up at our next ADIOS meeting.

Reply:
HIP supports unified virtual addressing, just like cuda (I'd consider cuda 4.0 historic by now, I don't think one has to worry about compatibility to pre cuda-4).

SYCL also supports something similar ("unified shared memory"). However, the native way in sycl, I guess, is to use sycl buffers -- but adios2 doesn't have an API to deal with buffers directly at this time, so one has to pass in pointers in any case.

Anyway, I'm not saying it's a bad idea to have the feature to specify a memory space explicitly -- but it'd be more application-friendly if the default memory space would be "auto" aka "UVA" or something, so that by default things just work, as they do in MPI.

Reply:
Yes I do agree that default usage would be nice (even though it will be a headache to implement) :) I'll discuss this with the others and figure out what we can do.